HW_4/task3.py

```diff
@@ -0,0 +1,116 @@
+# Задание 
+# Написать программу, которая скачивает изображения с заданных URL-адресов и сохраняет их на диск. Каждое
+# изображение должно сохраняться в отдельном файле, название которого соответствует названию изображения в URL-адресе.
+#  Например URL-адрес: https://example/images/image1.jpg -> файл на диске: image1.jpg
+#  Программа должна использовать многопоточный, многопроцессорный и асинхронный подходы.
+#  Программа должна иметь возможность задавать список URL-адресов через аргументы командной строки.
+#  Программа должна выводить в консоль информацию о времени скачивания каждого изображения и общем времени
+# выполнения программы.
+
+import os
+from pathlib import Path
+import requests
+import time
+import argparse
+import threading
+from multiprocessing import Process
+import asyncio
+import aiohttp
+
+images = []
+with open('images.txt', 'r') as f:
+    for image in f.readlines():
+        images.append(image.strip())
+
+PATH = Path('images')
+
+
+def download_img(url, dir_path=PATH):
+    start_time = time.time()
+    response = requests.get(url)
+    filename = url.split('/')[-1]
```

Comment:
Можно делить только 1 раз  
```suggestion
    filename = url.rsplit('/', 1)[-1]
```
или, что более универсально на других платформах
```suggestion
    filename = os.path.basename(url)
```